lib/realtime/server.ex

```diff
@@ -161,6 +164,15 @@ defmodule Realtime.Server do
     lookup({ets, subscription_key})
   end
 
+  defp update_subscription(server, {:search, _search_params} = subscription_key) do
+    {registry_key, ets} = GenServer.call(server, :subscription_info)
+    # Replace the old search subscription with the new one
+    Registry.unregister_match(Realtime.Registry, registry_key, {:search, %{}})
+    Registry.register(Realtime.Registry, registry_key, subscription_key)
```

Comment:
There is a [Registry.update_value/3](https://hexdocs.pm/elixir/1.6.6/Registry.html#update_value/3) function, but it isn't usable for our use case because we allow duplicate values. 

From the [unregister_match docs](https://hexdocs.pm/elixir/1.6.6/Registry.html#unregister/2):
> Unregisters all entries for the given key associated to the current process in registry.